pkg/operator/encryption/deployer/revisionedpod.go

```diff
@@ -89,6 +80,10 @@ func NewRevisionLabelPodDeployer(
 // DeployedEncryptionConfigSecret returns the deployed encryption config and whether all
 // instances of the operand have acknowledged it.
 func (d *RevisionLabelPodDeployer) DeployedEncryptionConfigSecret() (secret *corev1.Secret, converged bool, err error) {
+	if !d.HasSynced() {
```

Comment:
why do we need an informer here?
why we cannot have two listers instead?

usually, controllers are responsible for checking the caches have been sync before acting

Reply:
I don't like to fall over an non-synced informer here, reporting `converged=true` by accident. Better be defensive.